graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -524,7 +525,7 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
         if edge_traversal_is_folded:
             has_count_filter = has_fold_count_filter(context)
             _validate_fold_has_outputs_or_count_filter(
-                get_context_fold_info(context), has_count_filter, context['outputs'])
+                get_context_fold_info(context), has_count_filter, context['metadata'])
```

Comment:
`context['metadata']` is already available as `query_metadata_table` on line 400. This is absolutely not obvious at all since this function is quite large at this point, and I'll probably get around to refactoring it before too long...